tools/chplcheck/src/rule_types.py

```diff
@@ -185,6 +198,25 @@ def run_fixit_hooks(
                 fixits_from_hooks.append(f)
         return fixits_from_hooks
 
+    def get_settings_kwargs(self):
+        """
+        Returns a Dict of SettingName -> SettingValue for this rule
+        """
+
+        # find all relevant settings for this rule
+        # either they have no specific rule name, or they match this rule
+        setting_kwargs = {}
+        for setting in self.settings:
+            # default to None if the setting is not in the driver's settings
+            setting_kwargs[setting.setting_name] = None
+            # if setting is in the driver's settings, use that value
+            # otherwise, use the default value (ie pass nothing in the kwargs)
+            driver_setting = self.driver.config.rule_settings.get(setting)
+            if driver_setting is not None:
+                setting_kwargs[setting.setting_name] = driver_setting
```

Comment:
Since you default to `None` anyway, removing the condition and always running the code should not change anything, right? I thought you might be trying to do something about shadowing of global vs local settings with the same name, but the guaranteed ` = None` on line 211 would break that.

Reply:
ugh, no this old code